tree/forest/v7/inc/ROOT/RForest.hxx

```diff
@@ -87,10 +88,29 @@ private:
    std::unique_ptr<Detail::RPageSource> fSource;
 
 public:
-   static std::unique_ptr<RInputForest> Create(std::unique_ptr<RForestModel> model,
-                                               std::string_view forestName,
-                                               std::string_view storage);
-   static std::unique_ptr<RInputForest> Create(std::string_view forestName, std::string_view storage);
+   // Browse through the entries
+   class RIterator : public std::iterator<std::forward_iterator_tag, ForestSize_t> {
+   private:
+      using iterator = RIterator;
+      ForestSize_t fIndex = kInvalidForestIndex;
+   public:
+      RIterator() = default;
+      explicit RIterator(ForestSize_t index) : fIndex(index) {}
+      ~RIterator() = default;
+
+      iterator  operator++(int) /* postfix */        { auto r = *this; fIndex++; return r; }
```

Comment:
Using '++fIndex' might be 'slightly' better (if it is ever instantiated with something other than int/long.